server/register/custom-fields/__tests__/index.test.ts

```diff
@@ -0,0 +1,27 @@
+import { IStrapi } from "strapi-typed";
+import { registerCustomFields } from "..";
+
+describe("registerCustomFields()", () => {
+  it("should apply custom field when strapi supports it", () => {
+    const strapi: Pick<IStrapi, "customFields"> = {
+      customFields: {
+        register: jest.fn(),
+      },
+    };
+
+    registerCustomFields({ strapi } as any);
+
+    expect(strapi.customFields.register).toHaveBeenCalled();
+    expect((strapi.customFields.register as jest.Mock).mock.calls[0][0])
+      .toMatchInlineSnapshot(`
+      Object {
+        "name": "comments",
+        "plugin": "comments",
+        "type": "json",
+      }
+    `);
+  });
+  it("should not apply custom field when strapi does not supports it", () => {
+    expect(() => registerCustomFields({ strapi: {} } as any)).not.toThrow();
```

Comment:
Shouldn't be any alert provided, if custom fields are not suppported?

Reply:
We aim for gradual improvement. Custom file itself is not required to provide base functionality. Will log.warn be enough?